src/main/java/org/sonarlint/intellij/ui/nodes/LocalTaintVulnerabilityNode.kt

```diff
@@ -27,28 +27,53 @@ import org.sonarlint.intellij.finding.issue.vulnerabilities.LocalTaintVulnerabil
 import org.sonarlint.intellij.ui.tree.TreeCellRenderer
 import org.sonarlint.intellij.util.CompoundIcon
 import org.sonarsource.sonarlint.core.client.api.util.DateUtils
+import org.sonarsource.sonarlint.core.commons.ImpactSeverity
+import java.util.Collections
 import java.util.Locale
+import java.util.Map
 import java.util.Objects
 import javax.swing.Icon
 
 
 class LocalTaintVulnerabilityNode(val issue: LocalTaintVulnerability) : FindingNode(issue) {
 
     override fun render(renderer: TreeCellRenderer) {
-        val severity = StringUtil.capitalize(issue.severity().toString().lowercase(Locale.ENGLISH))
-        val type = issue.getType()
-        val typeStr = type.toString().replace('_', ' ').lowercase(Locale.ENGLISH)
-        renderer.setIconToolTip("$severity $typeStr")
+        val toolTipText: String
         val gap = if (JBUIScale.isUsrHiDPI()) 8 else 4
-        setIcon(
-            renderer,
-            CompoundIcon(
-                CompoundIcon.Axis.X_AXIS,
-                gap,
-                SonarLintIcons.type(type),
-                SonarLintIcons.severity(issue.severity())
+
+        if (issue.getCleanCodeAttribute() != null && issue.getImpacts().isNotEmpty()) {
+            val (quality, impact) = Collections.max(
+                issue.getImpacts().entries,
+                Map.Entry.comparingByValue(Comparator.comparing { obj: ImpactSeverity -> obj.ordinal })
+            )
+            val impactText = StringUtil.capitalize(impact.toString().lowercase())
+            val qualityText = quality.toString().lowercase()
+            toolTipText = "$impactText $qualityText"
```

Comment:
I have the impression we are doing 3 times the same thing: compute a displayable text out of impacts and quality. Do you think it would make sense to have this code in a single place?

Reply:
I don't see a proper way of doing this, the max value is also used for other things (get icon type), and those string attributes come from both LiveFinding and RuleTreeNode so it's difficult to factorize, and a utils class seems overkill, WDYT?